endaq/cloud/core.py

```diff
@@ -124,6 +125,58 @@ def get_file(self,
 
         return Dataset(f)
 
+    def download_all_ide_files(
+            self, output_directory: str = "", should_download_file_fn: Callable = lambda x: True, force_recompute_file_table: bool = False, file_limit: int = 100):
```

Comment:
Personally, I'd not define the lambda in the arguments, for the sake of how the docs will render. It might be clearer if you made it `should_download_file_fn: Optional[Callable] = None`, and then have an early line in the function do `should_download_file_fn = should_download_file_fn or (lambda x: True)`

Also, `should_download_file_fn` seems a bit long; how about just `filter_fn`? `force_recompute_file_table` is also long, and it isn't really computing anything, so maybe just `reload`?

Reply:
As for the lambda in the function definition, I'll make the change now.

As for the long parameter names, personally I prefer names which are longer but explain what they do without needing to look at a docstring, over short and not descriptive enough to know what it is  (though you have a point that it should be 'reload' not 'recompute'.

Reply:
I'd argue that the idea of a filter function is fairly well understood, so some variation of `filter_fn` works. I'm pretty sure I've seen that parameter name used elsewhere. nbd, though.